### PR TITLE
implement Collapsing AppBar

### PR DESCRIPTION
**Pull request by patrick-zhou-wattpad**
### Summary 
The task is to implement a collapsible Appbar. Initially, the current Appbar is in its default state. When the user scrolls up, and there are many interview items, the Appbar should transition to display only the topbar. This topbar should show the text "Hello Arighna" and a circular profile icon.

Screenshot 2023-11-07 at 10 59 37 AM

### Impacted Module(s)/Class(es)

presentation/ui/modules + components

### Type of Change(s)

- [ ] Bug Fix
- [x] New Feature
- [ ] Documentation
- [ ] Code Refactoring
- [ ] Kotlin Conversion
- [ ] Script Improvement
- [ ] Other: Please elaborate.
### How have I tested this ?

- [x] Manual Testing
- [ ] Unit Testing
- [ ] Other: Please elaborate

### Video Demo
https://github.com/Audienix/InterPrep/assets/144148115/0bff73cf-0a90-45c9-b767-62ac6b7b283e

### Known Issues
NA

### Clean Delivery Checklist

- [x] User interfaces are accessibility-enabled
- [x] No commented out code.
- [x] No IDE warnings in new files
- [x] No Detekt & Ktlint error.
- [ ] Unit tests for new code (for everything listed in [what we test](../blob/master/docs/best-practices/unit-tests.md#what-do-we-test))

### CODE REVIEW

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt [hunk 1]
```diff
@@ -89,7 +93,7 @@ fun DashboardScreen(
                     subtitle = "${stringResource(R.string.good)} ${getTimeOfDayGreeting()}",
                     todayInterviewList = todayInterviewList,
                     username = getNameInitials(dashboardViewModel.username),
-                    isInterviewDetailsVisible = true,
+                    isInterviewDetailsVisible = selectedInterviewList.value.size <=3 ||interviewListState.firstVisibleItemScrollOffset == 0,
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt]: We should not use it here, see the documentation

image

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/generic/IPLargeAppBar.kt [hunk 2]
```diff
@@ -69,7 +69,7 @@ fun IPLargeAppBar(
         horizontalAlignment = Alignment.Start,
         verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.dimension_4dp))
     ) {
-        GreetingsAndProfile(title, username, subtitle, onAvatarClick)
+        GreetingsAndProfile(title, username, if(isInterviewDetailsVisible)subtitle else "", onAvatarClick)
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/generic/IPLargeAppBar.kt]: if check can be done before this line and subtitle value can be passed here.

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt (hunk 1 above)
Comment (Audienix) [reply app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt]: There is actually an issue I am seeing with this code. See the video.

[Screen_recording_20231107_221922.webm](https://github.com/Audienix/InterPrep/assets/33220667/8204cb5e-49d0-4336-9136-4ccaee01bf74)

**Comment by Audienix** [comment]
Ideally we should use a custom Collapsible Appbar implementation but material 3 library Appbar doesn't give much flexibility now. So your way to handle this by checking visible item is good but we need to properly handle the scroll state and update the composable and also need to be mindful about avoiding too many recompositions. 

I found one great example to handle scroll state [here](https://stackoverflow.com/a/77198230) . Check this link, you may get the logic and implement it.

**Comment by patrick-zhou-wattpad** [comment]
Here is the updated version demo:

https://github.com/Audienix/InterPrep/assets/144148115/165409f8-20a3-4b12-a13e-f311b9f41f54

**Comment by patrick-zhou-wattpad** [comment]
Here is the updated version demo:
[collapse bar demo.webm](https://github.com/Audienix/InterPrep/assets/144148115/f907892f-c53d-4db6-9021-27f08c8dd5a0)

**Comment by Audienix** [comment]
> Here is the updated version demo: [collapse bar demo.webm](https://github.com/Audienix/InterPrep/assets/144148115/f907892f-c53d-4db6-9021-27f08c8dd5a0)

@patrick-zhou-wattpad I have changed scroll logic a bit to fix the issue with 4 items. Can you check and let me know it's fixed?
I have fixed. This is the demo:

https://github.com/Audienix/InterPrep/assets/144148115/107122b0-a61c-4324-8f02-dfcc8ebf06b6

**Review by Audienix — APPROVED** [review]

**Pull request merged**